### PR TITLE
Android: Add recursive game paths to UI

### PR DESCRIPTION
**Pull request by Ebola16**
Implements https://bugs.dolphin-emu.org/issues/10806

### CODE REVIEW

**Comment by JosJuice** [comment]
Your changes to `scanLibrary` are more complicated than needed. What you should do instead is make it so that you can pass a boolean parameter to `update` that's used instead of `false` on this line: https://github.com/dolphin-emu/dolphin/blob/master/Source/Android/jni/GameList/GameFileCache.cpp#L97

**Comment by JosJuice** [comment]
Assuming there is no simple way for us to automatically refresh the library when changing the setting, LGTM after fixing the lint issue.

**Comment by Ebola16** [comment]
Lint issue fixed. We could refresh the game library upon saving settings but I'd rather do that only if this setting is modified. I don't think there is a simple way to detect specific changed settings yet.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsActivityPresenter.java [hunk 1]
```diff
@@ -27,18 +31,22 @@
 
   private MenuTag menuTag;
   private String gameId;
+  private Context context;
+
+  private static final Set<String> modifiedSettings = new HashSet<>();
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsActivityPresenter.java:36]: Would it be possible to make this non-static? You could pass it to `saveSettings` as an argument.

Comment (Ebola16) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsActivityPresenter.java]: Yes, done.

**Pull request merged**